pkg/api/v1alpha1/work_types.go

```diff
@@ -0,0 +1,156 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// WorkSpec defines the desired state of Work
+type WorkSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Workload represents the manifest workload to be deployed on spoke cluster
+	Workload WorkloadTemplate `json:"workload,omitempty"`
+}
+
+// WorkloadTemplate represents the manifest workload to be deployed on spoke cluster
+type WorkloadTemplate struct {
+	// Manifests represents a list of kuberenetes resources to be deployed on the spoke cluster.
+	// +optional
+	Manifests []runtime.RawExtension `json:"manifests,omitempty" protobuf:"bytes,2,rep,name=manifests"`
```

Comment:
Is `runtime.RawExtension` going to play nice with webhooks (i.e. disabling unknown fields)?

Reply:
i do not think so, it will only contain raw bytes. So in the webhook, it needs to be decoded and typed